File: DMOJ/CCC/SurmisingASprintersSpeed.py
# CCC '20 S1 - Surmising a Sprinter's Speed

import sys

input = sys.stdin.readline

arr = [list(map(int, input().split())) for i in range(int(input()))]
arr.sort()
ans = 0
for i in range(len(arr) - 1):
    speed = abs((arr[i][1] - arr[i + 1][1]) / (arr[i][0] - arr[i + 1][0]))
    ans = max(ans, speed)
print(ans)

File: DMOJ/CCC/SurmisingASprintersSpeed.cpp
// CCC '20 S1 - Surmising a Sprinter's Speed

#include <bits/stdc++.h>
using namespace std;
#define m second
#define t first

int n;
double ans = 0;
vector<pair<double, double>> arr;
int main(){
    ios_base::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);
    cin >> n;
    while (n--){
        double a, b; cin >> a >> b;
        arr.push_back({a, b});
    }
    sort(arr.begin(), arr.end());
    for (int i=0; i<arr.size()-1; i++){
        pair<double, double> a = arr[i], b = arr[i+1];
        double speed = abs((double)(a.m-b.m)/(double)(a.t-b.t));
        ans = max(ans, speed);
    }
    cout << ans << '\n';

    return 0;
}